run_infrastructure.py

```diff
@@ -0,0 +1,234 @@
+"""
+...
+"""
+import argparse
+import os
+import sys
+from typing import (
+    List,
+)
+
+DOCKER_WAIT = 15
+DOCKERFILE_DATASTORES = [
+    "postgres",
+    "mysql",
+    "mongodb",
+    "mssql",
+]
+EXTERNAL_DATASTORE_CONFIG = {
+    "snowflake": ["SNOWFLAKE_TEST_URI"],
+    "redshift": ["REDSHIFT_TEST_URI", "REDSHIFT_TEST_DB_SCHEMA"],
+}
+EXTERNAL_DATASTORES = list(EXTERNAL_DATASTORE_CONFIG.keys())
+IMAGE_NAME = "fidesops"
+
+
+def run_infrastructure(
+    datastores: List[str] = [],  # Which infra should we create? If empty, we create all
+    open_shell: bool = False,  # Should we open a bash shell?
+    pytest_path: str = "",  # Which subset of tests should we run?
+    run_application: bool = False,  # Should we run the Fidesops webserver?
+    run_quickstart: bool = False,  # Should we run the quickstart command?
+    run_tests: bool = False,  # Should we run the tests after creating the infra?
+) -> None:
+    """
+    - Create a Docker Compose file path for all datastores specified in `datastores`.
+    - Defaults to creating infrastructure for all datastores in `DOCKERFILE_DATASTORES` if none
+    are provided.
+    - Optionally runs integration tests against those datastores from the container identified
+    with `IMAGE_NAME`.
+    """
+
+    if len(datastores) == 0:
+        os.system(
+            f'echo "no datastores specified, configuring infrastructure for all datastores"'
+        )
+        datastores = DOCKERFILE_DATASTORES + EXTERNAL_DATASTORES
+    else:
+        os.system(f'echo "datastores specified: {", ".join(datastores)}"')
+
+    # Configure docker-compose path
+    path: str = get_path_for_datastores(datastores)
```

Comment:
maybe we also modify datastores to be `set(datastores)` to make sure the same datastore isn't passed in twice -